### ISSUE TITLE
When Opened

### ISSUE THREAD

**Issue by callumb** [spatie/opening-hours#130]
Is there a way of getting when the store opened? 

We have two opening times: One for collection and one for deliveries. 

So stay collection was: 10:00-14:00 then 17:00 - 23:00.
and delivery was: 18:00 - 23:00.

The current time is 20:04 I want to get the 17:00 based on 20:04 or if the time was 12:34 then it would return 10:00 and if not open then return false.

**Comment by kylekatarnls** [comment]
Hi, this would need a big rearrangement of our code to implement `previousOpen` and `previousClose` that would allow you to do:

```php
$date = $openingHours->isClosed() ? false : $openingHours->previousOpen();
```

It will take times, but it's interesting for the next version.

**Comment by kylekatarnls** [comment]
With version 2.6.0, you will be able to do:

```php
$now = new DateTime('now');
$range = $openingHours->currentOpenRange($now);

if ($range) {
    echo "It's open since ".$range->start()."\n";
    echo "It will close at ".$range->end()."\n";
} else {
    echo "It's closed, it will open at ".$openingHours->nextOpen($now)->format('l H:i');
}

$date = $openingHours->currentOpenRangeStart($now);

if ($date) {
    echo "It's open since ".$date->format('H:i');
} else {
    echo "It's closed";
}

$date = $openingHours->currentOpenRangeStart($now);

if ($date) {
    echo "It will close at ".$date->format('H:i');
} else {
    echo "It's closed";
}
```

**Comment by callumb** [comment]
Thats amazing! Thanks for sorting it out and so quickly.

**Comment by callumb** [comment]
Also do you have a eta on when it will be pushed to composer?

**Comment by kylekatarnls** [comment]
You can use right now `"spatie/opening-hours": "dev-master"` to test it.

**Comment by kylekatarnls** [comment]
Available now in 2.6.0:
https://github.com/spatie/opening-hours/releases/tag/2.6.0
